tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckDates.java

```diff
@@ -654,6 +662,70 @@ && isTooMuchWiderThan(value, abbrValue)
         return this;
     }
 
+    private void checkTimeFormatMatchesRegion(String value, List<CheckStatus> result) {
+        DateTimePatternGenerator dtpg = DateTimePatternGenerator.getEmptyInstance();
+        Map<String /* region */, PreferredAndAllowedHour> timeData = sdi.getTimeData();
+        Map<String, String> likelySubtags = sdi.getLikelySubtags();
+        String localeID = getResolvedCldrFileToCheck().getLocaleID();
+        String jPattern = getRegionHourFormat(timeData, localeID, likelySubtags);
+        if (jPattern == null) {
+            CheckStatus item =
+                    new CheckStatus()
+                            .setCause(this)
+                            .setMainType(CheckStatus.errorType)
+                            .setSubtype(Subtype.inconsistentTimePattern)
+                            .setMessage("No hour format found");
+            result.add(item);
+            return;
+        }
+        String shortPatSkeleton = dtpg.getBaseSkeleton(value); // e.g., "ahm" or "Hm"
+        String jPatSkeleton = dtpg.getBaseSkeleton(jPattern); // e.g., "ah" or "H"
+        final char[] timeCycleChars = {'H', 'h', 'K', 'k'};
+        for (char timeCycleChar : timeCycleChars) {
+            if (jPatSkeleton.indexOf(timeCycleChar) >= 0
+                    && shortPatSkeleton.indexOf(timeCycleChar) < 0) {
+                String message =
+                        "Time format does not match region; expected "
+                                + timeCycleChar
+                                + " in the value "
+                                + value;
+                CheckStatus item =
+                        new CheckStatus()
+                                .setCause(this)
+                                .setMainType(CheckStatus.errorType)
```

Comment:
A way to have this be an error in SurveyTool but only a warning in ConsoleCheck (so you can commit) was suggested to me by Steven when I had the same problem in a different PR, see lines 110-119 here: https://github.com/unicode-org/cldr/pull/4547/files

Reply:
Good to know -- I ended up using the "-1" option for "singleThread" to avoid crashing without "-e". I'm assuming for now at least we want warnings rather than errors.